src/docs/sphinx/blueprint_mesh.rst

```diff
@@ -1254,59 +1263,101 @@ Outputting Meshes for Visualization
-    conduit::relay::io_blueprint::save(const conduit::Node &mesh,
-                                       const std::string &path,
-                                       const std::string &protocol);
+Save a mesh to disk using a specific protocol:
 
-It's important to note that both of these functions expect the given path to have
-a valid extension to properly output results. The valid extensions for these
-functions are as follows:
+.. code:: cpp
 
-- ``.blueprint_root`` (JSON Extension)
-- ``.blueprint_root_hdf5`` (HDF5 Extension)
+    conduit::relay::io::blueprint::write_mesh(const conduit::Node &mesh,
+                                              const std::string &protocol,
+                                              const std::string &path);
 
-Files output from these functions can be opened and subsequently visualized
-directly using `VisIt <https://visit.llnl.gov>`_.
+Save a mesh to disk using a specific protocol and options:
+
+.. code:: cpp
+
+    /// Options accepted via the `opts` Node argument:
+    ///
+    ///      file_style: "default", "root_only", "multi_file"
+    ///            when # of domains == 1,  "default"   ==> "root_only"
+    ///            else,                    "default"   ==> "multi_file"
+    ///
+    ///      suffix: "default", "cycle", "none" 
+    ///            when # of domains == 1,  "default"   ==> "off"
+    ///            else,                    "default"   ==> "cycle"
+    ///
+    ///      mesh_name:  (used if present, default ==> "mesh")
+    ///
+    ///      number_of_files:  {# of files}
+    ///            when "multi_file":
+    ///                 <= 0, use # of files == # of domains
+    ///                  > 0, # of files == number_of_files
+    ///
+    conduit::relay::io::blueprint::write_mesh(const conduit::Node &mesh,
+                                              const std::string &path,
+                                              const std::string &protocol,
+                                              const conduit::Node &opts);
+
+Loading Meshes from Files
+==========================
+
+If you have a mesh written to a set of blueprint files, you can load them by 
+passing the root file path to the following ``conduit::relay::io::blueprint``
+library functions:
+
+Load a mesh given a root file:
+
+.. code:: cpp
+
+    conduit::relay::io::blueprint::read_mesh(const std::string &root_file_path,
+                                             conduit::Node &mesh);
+
+
+Load a mesh given a root file and options:
+
+.. code:: cpp
+
+    /// Options accepted via the `opts` Node argument:
+    ///
+    ///      mesh_name: "{name}"
+    ///          provide explicit mesh name, for cases where bp data includes
+    ///           more than one mesh.
+    ///
+    conduit::relay::io::blueprint::read_mesh(const std::string &root_file_path,
+                                             const conduit::Node &opts,
+                                             conduit::Node &mesh);
```

Comment:
I have a habit of using input nodes as the first arg, and output nodes as the final arg. 

That said, I don't think I am completely consistent.  This also causes confusion with the python interface, where since there are kwargs and differente considerations related to those, the order can be differente between the C++ and Python  interfaces.